### DISCUSSION TITLE
Installation problems on Linux Mint

### DISCUSSION THREAD

**Discussion by quiet-ranger** [danielmiessler/Fabric#273] [Q&A]
First of all, based on the video and textual description, this looks like a very interesting project to look into and perhaps even contribute to.

I run into a fair number of obstacles trying to install fabric and the biggest was the requirement to run on Python 10.  Eventually I got there and, in fact, I Dockerized the installation so that it stays stable and does not interfere with anything else (I may consider making the Docker image public if there is enough interest).  I even got all the right API keys installed but...when I tried my very first prompt...the system asked me to pay OpenAI for access to ChatGPT 4.  This should be clearly stated in the requirements upfront so that people don't waste time. 

On the other hand, I would like to ask how easy would it be to add a switch to revert to ChatGPT 3 or 3.5 for those who are not in a position to fork money for a bit of tinckering.  Has anyone tried it?  Did anyone succeed?

**Comment by waldo323** [comment]
For fabric to use ChatGPT (or Claude), it needs to use the API which is only available for paid users. 

The option for using local models using ollama is an alternative which doesn't have a subscription fee and gives you flexibility regarding which model or models you want to use.  ollama can also be run on a separate machine which you can access over the network.

**Reply by xssdoctor** [reply]
It can already run local Ollama models. --model or --chanfeDefaultModel. For a list of available models run --listmodels
